### PR TITLE
FL API docs

### PR DESCRIPTION
**Pull request by fuhuifang**
Instructions on how to use FL APIs.

### CODE REVIEW

File: docs/pipeline/federated-learning-doc.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+```
+![image](img/fl-diagram.png)
+
+## Configure your pipeline through a yaml file
+Example: [demo_federated_learning.yaml](https://github.com/Azure/shrike/blob/main/examples/pipelines/config/experiments/demo_federated_learning.yaml)
+
+You need to add a `federated_config` section to your pipeline yaml file:
+```yaml
+federated_config:
+  data_transfer_component: office.smart_reply.fl.data_transfer
```
Comment (dkmiller) [inline docs/pipeline/federated-learning-doc.md:28]: Suggestion: remove 1P-specific team names (Office, Smart Reply) and do something more like `contoso.my_organization.fl.data_transfer`.

File: docs/pipeline/federated-learning-doc.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        dataset: test_csv
+  params:
+    msg: "random msg"
+```
+To use the parameters in a pipeline, you can use the dot path. For example, `config.federated_config.params.msg` will return "random msg". Note that for `train` in silo, you can refer to per-silo parameters directly. For example, `silo.params.dataset` returns "test_uploaded_from_local", "test_csv", and "test_csv" for silo1, silo2, and silo3 respectively.
+
+## Write the Python script for your pipeline
+Example: [demo_federated_learning.py](https://github.com/Azure/shrike/blob/main/examples/pipelines/experiments/demo_federated_learning.py)
+
+Unlike a typical pipeline script, here you need to create a class inheriting from `shrike.pipeline.FederatedPipelineBase` and **implement the four required steps**: `preprocess`, `train`, `midprocess`, and `postprocess`.
```
Comment (dkmiller) [inline docs/pipeline/federated-learning-doc.md:98]: Not all steps are required, right? Hopefully `preprocess` and `postprocess` (at least) are optional?

**Review by dkmiller — CHANGES_REQUESTED** [review]
Looks pretty good, just made some minor comments.

One other thing, from the message

```
INFO     -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
  - compliant_logging/per-file-logging.md
  - rfc/federated-learning.md
 ```

in the [docs build](https://github.com/Azure/shrike/runs/4478248435?check_suite_focus=true), it looks like you need to add both those docs to `mkdocs.yml`.

File: docs/pipeline/federated-learning-doc.md (hunk 2 above)
Comment (fuhuifang) [reply docs/pipeline/federated-learning-doc.md:98]: Unfortunately all steps are required for now. I've added it to the to-do list: https://github.com/Azure/shrike/issues/258. 

Is it okay if I keep doc as-is, and fix it in the PR updating the APIs?

Comment (dkmiller) [reply docs/pipeline/federated-learning-doc.md:98]: Absolutely, keep as-is. Docs should reflect reality :)

**Review by dkmiller — APPROVED** [review]

**Pull request merged**